app/src/org/commcare/android/analytics/PrefStats.java

```diff
@@ -0,0 +1,90 @@
+package org.commcare.android.analytics;
+
+import android.content.SharedPreferences;
+import android.util.Base64;
+import android.util.Log;
+
+import org.commcare.dalvik.application.CommCareApp;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+/**
+ * Generic logic for storing stats/logs in app preference storage layer.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class PrefStats {
```

Comment:
Initially I was going to store all this in the preferences storage layer, but Clayton then told me afterwards that I should change it to mimic how normal logs are stored, in an sql db.
I left the generalizing PrefStats refactors in the PR since they might be useful in the future.